lib/index.ts

```diff
@@ -4,6 +4,7 @@ import * as commander from "commander";
 import * as fs from "fs";
 import * as path from "path";
 import * as ts from "typescript";
+var glob = require("glob");
```

Comment:
Let's use `import` as for other imports.

Reply:
@rattkin

```
git checkout master
git rebase upstream/master
git push --force
```

Reply:
```
>git checkout master 
Previous HEAD position was a52eef1 Merge pull request #31 from zenflow/patch-3
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
```

```
>git rebase upstream/master                                               
Successfully rebased and updated refs/heads/master.
```

```
>git push --force
Enumerating objects: 14, done.
Counting objects: 100% (14/14), done.
Delta compression using up to 8 threads
Compressing objects: 100% (6/6), done.
Writing objects: 100% (10/10), 1.56 KiB | 1.56 MiB/s, done.
Total 10 (delta 6), reused 4 (delta 4), pack-reused 0      
remote: Resolving deltas: 100% (6/6), completed with 3 local objects.
To https://github.com/rattkin/ts-interface-builder.git
 + 83066f7...3178e11 master -> master (forced update) 
```

thanks always wanted to try force push

Reply:
@rattkin Don't forget to address the initial comment here: "Let's use `import` as for other imports." https://github.com/gristlabs/ts-interface-builder/pull/35/files#r528726650

Reply:
@zenflow Yes I forgot completely.

Reply:
I can even do it like in Angular:
import { sync } from "glob"; 

Either way, my lint has problems with:
```
const globFiles = [].concat(...files.map(p => sync(p)));

No overload matches this call.
  Overload 1 of 2, '(...items: ConcatArray<never>[]): never[]', gave the following error.
    Argument of type 'string[]' is not assignable to parameter of type 'ConcatArray<never>'.
      The types returned by 'slice(...)' are incompatible between these types.
        Type 'string[]' is not assignable to type 'never[]'.
          Type 'string' is not assignable to type 'never'.
  Overload 2 of 2, '(...items: ConcatArray<never>[]): never[]', gave the following error.
    Argument of type 'string[]' is not assignable to parameter of type 'ConcatArray<never>'.
```


However this works:
```
let blank:string[] = []
  const globFiles = blank.concat(...files.map(p => sync(p)));
```
But the tests are very broken now. I have no idea what to do!:
```
2) ts-interface-builder
       should strip out promises:

      AssertionError [ERR_ASSERTION]: Expected values to be loosely deep-equal:

'import * as t from "ts-interface-checker";\n' +
  '// tslint:disable:object-literal-key-quotes\n' +
  '\n' +
  'export const ILRUCache = t.iface([], {\n' +
  '  "capacity": "number",\n' +
  '  "isReady": "boolean",\n' +
  '  "set": t.func("boolean", t.param("item", "ICacheItem"), t.param("overwrite", "boolean", true)),\n' +
  '  "get": t.func("ICacheItem", t.param("key", "string")),\n' +
  '});\n' +
  '\n' +
  'const exportedTypeSuite: t.ITypeSuite = {\n' +
  '  ILRUCache,\n' +
  '};\n' +
  'export defa...

should loosely deep-equal

'import * as t from "ts-interface-checker";\r\n' +
  '// tslint:disable:object-literal-key-quotes\r\n' +
  '\r\n' +
  'export const ILRUCache = t.iface([], {\r\n' +
  '  "capacity": "number",\r\n' +
  '  "isReady": "boolean",\r\n' +
  '  "set": t.func("boolean", t.param("item", "ICacheItem"), t.param("overwrite", "boolean", true)),\r\n' +
  '  "get": t.func("ICacheItem", t.param("key", "string")),\r\n' +
  '});\r\n' +
  '\r\n' +
  'const exportedTypeSuite: t.ITypeSuite = {\r\n' +
  '  ILRUCache,\r\n' +
 ...
      + expected - actual

      -import * as t from "ts-interface-checker";
      -// tslint:disable:object-literal-key-quotes
      -
      -export const ILRUCache = t.iface([], {
      -  "capacity": "number",
      -  "isReady": "boolean",
      -  "set": t.func("boolean", t.param("item", "ICacheItem"), t.param("overwrite", "boolean", true)),
      -  "get": t.func("ICacheItem", t.param("key", "string")),
      -});
      -
      -const exportedTypeSuite: t.ITypeSuite = {
      -  ILRUCache,
      -};
      -export default exportedTypeSuite;
      +import * as t from "ts-interface-checker";
      +// tslint:disable:object-literal-key-quotes
      +
      +export const ILRUCache = t.iface([], {
      +  "capacity": "number",
      +  "isReady": "boolean",
      +  "set": t.func("boolean", t.param("item", "ICacheItem"), t.param("overwrite", "boolean", true)),
      +  "get": t.func("ICacheItem", t.param("key", "string")),
      +});
      +
      +const exportedTypeSuite: t.ITypeSuite = {
      +  ILRUCache,
      +};
      +export default exportedTypeSuite;

      at C:\Users\Jan Müller\Documents\development\ts-interface-builder\test\test_index.ts:18:12
      at Generator.next (<anonymous>)
      at fulfilled (test\test_index.ts:5:58)
```
and more

Reply:
> I can even do it like in Angular:
> import { sync } from "glob";

@rattkin I suggest instead `import * as glob from "glob";` so that when you call `glob.sync()` the code says what it's doing (and not just that it's doing something synchronously).

The issue with failing tests looks like an ["autocrlf"](https://docs.github.com/en/free-pro-team@latest/github/using-git/configuring-git-to-handle-line-endings) issue. Basically if this setting is enabled (which it is by default) then when checking out code on Windows, Git will change all line endings in all files from `\n` (Linux/Unix/Mac) to `\r\n` (Windows). In my experience this usually creates more problems than it solves. This is a good example. The generated code uses `\n` line endings but the files defining the expected output have had line endings changed to `\r\n` by Git, so it looks like the actual generated code does not match the expected output. I suggest disabling autocrlf globally (google it) so you're always working with the exact same files that were checked in, and just when you create a new file make sure it uses the right line endings for the project (usually `\n`). After you disable autocrlf (either globally, or for this specific project), you can fix the files in your working copy with `git reset --hard HEAD` (note that this will also wipe out any uncommitted changes).

Reply:
> Either way, my lint has problems with: [...]
> However this works: [...]

@rattkin Good problem solving. I would inline the blank/dummy/temp variable like so:
```ts
const globFiles = ([] as string[]).concat(...files.map(p => sync(p)));
```